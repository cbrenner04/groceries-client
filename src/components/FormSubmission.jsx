import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

// TODO: check react-bootstrap has some way of handling the grid
const FormSubmission = (props) => (
  <div className="d-grid gap-2 mt-3">
    <Button type="submit" variant="success" disabled={props.disabled || false}>
      {props.submitText}
    </Button>
    {props.displayCancelButton && (
      <Button variant="link" onClick={props.cancelAction || (() => undefined)}>
        {props.cancelText || ''}
      </Button>
    )}
  </div>
);

FormSubmission.propTypes = {
  disabled: PropTypes.bool,
  submitText: PropTypes.string.isRequired,
  displayCancelButton: PropTypes.bool,
  cancelAction: PropTypes.func,
  cancelText: PropTypes.string,
};

export default FormSubmission;
