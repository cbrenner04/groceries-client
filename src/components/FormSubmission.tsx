import React, { type MouseEventHandler } from 'react';
import { Button } from 'react-bootstrap';

interface IFormSubmissionProps {
  disabled?: boolean;
  submitText: string;
  displayCancelButton: boolean;
  cancelAction: MouseEventHandler;
  cancelText?: string;
}

const FormSubmission: React.FC<IFormSubmissionProps> = ({
  disabled = false,
  submitText,
  displayCancelButton,
  cancelAction,
  cancelText = '',
}): React.JSX.Element => (
  <div className="d-grid gap-2 mt-3">
    <Button type="submit" variant="success" disabled={disabled}>
      {submitText}
    </Button>
    {displayCancelButton && (
      <Button variant="link" onClick={cancelAction}>
        {cancelText}
      </Button>
    )}
  </div>
);

export default FormSubmission;
