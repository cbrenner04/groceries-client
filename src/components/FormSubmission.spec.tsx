import React from 'react';
import { render } from '@testing-library/react';

import FormSubmission from './FormSubmission';

describe('FormSubmission', () => {
  it('renders the spinner', () => {
    const { container, getAllByRole } = render(
      <FormSubmission submitText="foo" displayCancelButton={true} cancelAction={jest.fn()} cancelText="bar" />,
    );

    expect(container).toMatchSnapshot();
    expect(getAllByRole('button')[0]).toHaveTextContent('foo');
    expect(getAllByRole('button')[1]).toHaveTextContent('bar');
  });
});
