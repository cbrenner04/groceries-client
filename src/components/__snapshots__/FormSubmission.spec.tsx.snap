// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSubmission renders the spinner 1`] = `
<div>
  <div
    class="d-grid gap-2 mt-3"
  >
    <button
      class="btn btn-success"
      type="submit"
    >
      foo
    </button>
    <button
      class="btn btn-link"
      type="button"
    >
      bar
    </button>
  </div>
</div>
`;
